packages/core/src/CSL/cslToCore/certificate.ts

```diff
@@ -34,52 +35,54 @@ const stakeDelegation = (certificate: CSL.StakeDelegation): StakeDelegationCerti
-    const relay = relays.get(i);
-    const relayByAddress = relay.as_single_host_addr();
-    const relayByName = relay.as_single_host_name();
-    const relayByNameMultihost = relay.as_multi_host_name();
+const createCardanoRelays = (relays: CSL.Relays): Relay[] =>
+  usingAutoFree((scope) => {
+    const result: Relay[] = [];
+    for (let i = 0; i < relays.len(); i++) {
+      const relay = scope.manage(relays.get(i));
+      const relayByAddress = relay.as_single_host_addr();
+      const relayByName = relay.as_single_host_name();
+      const relayByNameMultihost = relay.as_multi_host_name();
 
-    if (relayByAddress) {
-      // RelayByAddress
-      result.push({
-        __typename: 'RelayByAddress',
-        ipv4: relayByAddress.ipv4()?.ip().join('.'),
-        ipv6: relayByAddress.ipv6()?.ip().join('.'),
-        port: relayByAddress.port()
-      });
+      if (relayByAddress) {
+        // RelayByAddress
+        result.push({
+          __typename: 'RelayByAddress',
+          ipv4: relayByAddress.ipv4()?.ip().join('.'),
+          ipv6: relayByAddress.ipv6()?.ip().join('.'),
+          port: relayByAddress.port()
+        });
+      }
+      if (relayByName) {
+        // RelayByName
+        result.push({
+          __typename: 'RelayByName',
+          hostname: relayByName.dns_name().record(),
+          port: relayByName.port()
+        });
+      }
+      if (relayByNameMultihost) {
+        // RelayByNameMultihost
+        result.push({
+          __typename: 'RelayByNameMultihost',
+          dnsName: relayByNameMultihost.dns_name().record()
+        });
+      }
     }
-    if (relayByName) {
-      // RelayByName
-      result.push({
-        __typename: 'RelayByName',
-        hostname: relayByName.dns_name().record(),
-        port: relayByName.port()
-      });
-    }
-    if (relayByNameMultihost) {
-      // RelayByNameMultihost
-      result.push({
-        __typename: 'RelayByNameMultihost',
-        dnsName: relayByNameMultihost.dns_name().record()
-      });
-    }
-  }
-  return result;
-};
+    return result;
+  });
 
-const createCardanoOwners = (owners: CSL.Ed25519KeyHashes, networkId: NetworkId): RewardAccount[] => {
-  const result: RewardAccount[] = [];
-  for (let i = 0; i < owners.len(); i++) {
-    const keyHash = owners.get(i);
-    const stakeCredential = CSL.StakeCredential.from_keyhash(keyHash);
-    const rewardAccount = CSL.RewardAddress.new(networkId, stakeCredential);
-    result.push(RewardAccount(rewardAccount.to_address().to_bech32()));
-  }
-  return result;
-};
+const createCardanoOwners = (owners: CSL.Ed25519KeyHashes, networkId: NetworkId): RewardAccount[] =>
+  usingAutoFree((scope) => {
+    const result: RewardAccount[] = [];
+    for (let i = 0; i < owners.len(); i++) {
+      const keyHash = scope.manage(owners.get(i));
+      const stakeCredential = CSL.StakeCredential.from_keyhash(keyHash);
+      const rewardAccount = CSL.RewardAddress.new(networkId, stakeCredential);
```

Comment:
Those probably needs to be managed

Reply:
214d62f11644c689a1e38c3f1811f88e87acf8ac